### PR TITLE
[terra-form-select] Dialog Modal didn't close with 'Esc' key press.

### PR DESCRIPTION
**Pull request by AH106586Harika**
### Summary
Fixed - Dialog Modal didn't close with 'Esc' key press.

**What was changed:**
Added a condition to manage empty input text when the combo box is in focus.

**Why it was changed:**
Expected behaviour is to close the Modal with a single press of the 'esc' key.
### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

[UXPLATFORM-10076](https://jira2.cerner.com/browse/UXPLATFORM-10076)

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-form-select/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 ## Unreleased
 
 ## 6.52.0 - (December 11, 2023)
+* Added
```
Comment (supreethmr) [inline packages/terra-form-select/CHANGELOG.md:6]: need to add line break for consistency

Comment (AH106586Harika) [reply packages/terra-form-select/CHANGELOG.md:7]: Done.

**Review by supreethmr — APPROVED** [review]

**Comment by sugan2416** [comment]
@AH106586Harika In this example, we are able to close dialog with single Esc press. When modal is opened and form-select is opened, first Esc press closes the dropdown and second one closes the modal. This is same as defined in test: https://github.com/cerner/terra-framework/blob/370d1e76d029eeea14cdd5a26238d1de46d2cb06/packages/terra-dialog-modal/tests/wdio/dialog-modal-spec.js#L71 Can we verify this

**Comment by sugan2416** [comment]
Can we add evidence of dialog closure with this change

**Comment by AH106586Harika** [comment]
> Can we add evidence of dialog closure with this change


https://github.com/cerner/terra-core/assets/126655256/9aaa1622-8e2f-4817-8c4a-9b729eb471c5

**Review by sugan2416 — APPROVED** [review]

**Pull request merged**